saleor/graphql/discount/dataloaders.py

```diff
@@ -109,16 +110,32 @@ def batch_load(self, keys):
         return [vouchers.get(voucher_id) for voucher_id in keys]
 
 
-class VoucherByCodeLoader(DataLoader):
-    context_key = "voucher_by_code"
+class VoucherCodeByCodeLoader(DataLoader):
+    context_key = "voucher_code_by_code"
 
     def batch_load(self, keys):
-        vouchers = (
-            Voucher.objects.using(self.database_connection_name)
+        voucher_codes = (
+            VoucherCode.objects.using(self.database_connection_name)
+            .select_related("voucher")
```

Comment:
Do we need this `select_related`?

Reply:
In line `351` we create `CheckoutInfo` and we need pass `voucher` instance into, so we avoiding extra queries using `select_related` here